language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/common/utils/CommonUtil.java

```diff
@@ -1017,6 +1017,16 @@ public static boolean isValidIdentifier(String identifier) {
         return identifier.matches(PatternConstants.IDENTIFIER_PATTERN);
     }
 
+    /**
+     * Escapes the escape characters present in an identifier.
+     *
+     * @param identifier Identifier
+     * @return The identifier with escape characters escaped
+     */
+    public static String escapeEscapeCharsInIdentifier(String identifier) {
+        return identifier.replaceAll("\\\\", "\\\\\\\\");
```

Comment:
Shall we make sure whether this is a client specific behaviour? This is because, it shows the actual value when we set it as the label, and only revert it in the insert text.

Reply:
I checked this. This affects when the text edit is being applied. Even when we perform a workspace edit, this issue is not there. In hover (documentation) this issue is not there for user defined types because those types are wrapped with quotes (in markdown). WDYT?

Reply:
+1 to proceed then. We can fix this when we move to textEdit based auto completions.